### PR TITLE
DS Error validator

### LINKED ISSUE #118: How to handleDS.errors?
Is there any code need be wriiten if server throws 422 errors.The one which is mentioned in the below link.

http://emberjs.com/api/data/classes/DS.Errors.html

### PR DESCRIPTION
**Pull request by offirgolan**
Creates a link between this library and Ember-Data's [DS.Errors](http://emberjs.com/api/data/classes/DS.Errors.html) to fetch the latest message for the given attribute.

Resolves #118 
 
```javascript
 validator('ds-error');
```

### CODE REVIEW

**Comment by offirgolan** [comment]
@stefanpenner @rwjblue @blimmer open for review

File: app/validators/ds-error.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright 2015, Yahoo! Inc.
```
Comment (blimmer) [inline app/validators/ds-error.js]: idk if you need to bump this up for legal stuffs

File: app/validators/ds-error.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+ *  @class DS Error
+ *  @module Validators
+ *  @extends Base
+ */
+export default Base.extend({
+  validate(value, options, model, attribute) {
+    let errors = get(model, 'errors');
+
+    if(!isNone(errors) && errors instanceof DS.Errors && errors.has(attribute)) {
+      return get(errors.errorsFor(attribute), 'lastObject.message');
```
Comment (blimmer) [inline app/validators/ds-error.js]: I think it makes sense to get the `lastObject` in the errors array, but we should definitely document that behavior so people don't get confused? Do you think we need to expose a way to get all the errors?

File: app/validators/ds-error.js (hunk 1 above)
Comment (offirgolan) [inline app/validators/ds-error.js]: Ooooh yeah, Ill do it in a separate commit. Thanks for the reminder :D

File: app/validators/ds-error.js (hunk 2 above)
Comment (offirgolan) [inline app/validators/ds-error.js]: Look at the comment above. `...fetch the latest message for the given attribute.`

Comment (blimmer) [inline app/validators/ds-error.js]: Ah, I forgot we're using the autogen-ed docs. Looks good.

Comment (offirgolan) [inline app/validators/ds-error.js]: :+1:
